src/NodaTime/DateInterval.cs

```diff
@@ -167,5 +186,70 @@ public void Deconstruct(out LocalDate start, out LocalDate end)
             start = Start;
             end = End;
         }
+
+        /// <summary>
+        /// Checks whether the given interval has days in common with the current interval.
+        /// </summary>
+        /// <param name="interval">The <see cref="DateInterval"/> to check for intersection with this interval.</param>
+        /// <returns>
+        /// <c>true</c> if <paramref name="interval"/> has days in common with this interval; <c>false</c> otherwise.
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        /// The start and end dates of <paramref name="interval" /> are not in the same calendar
+        /// as the start and end date of this interval.
+        /// </exception>
+        public bool Intersects([NotNull]DateInterval interval)
+        {
+            Preconditions.CheckNotNull(interval, nameof(interval));
+
+            Preconditions.CheckArgument(
```

Comment:
It may well be worth adding:

- A private helper method to assert that an interval is non-null and in the same calendar system
- A `Calendar` property which just returns `Start.Calendar`

Reply:
Yeah, I thought of doing something like that, but ended up not doing.